src/mpp/Dec.hpp

```diff
@@ -82,6 +82,10 @@ constexpr auto detectFamily()
 		      "Can't decode to constant type");
 	if constexpr (is_wrapped_family_v<T>) {
 		return family_sequence<T::family>{};
+	} else if constexpr (tnt::is_optional_v<U>) {
+		return family_sequence_populate<compact::MP_NIL>(detectFamily<tnt::value_type_t<U>>());
```

Comment:
Nit: what's the advantage over

```suggestion
		return family_sequence<compact::MP_NIL, detectFamily<tnt::value_type_t<U>>()>{};
```
?

Reply:
The advantage is that it works :)
`detectFamily` returns a `family_sequence` object, which cannot be a template parameter for another `family_sequence`.